expression/builtin_regexp.go

```diff
@@ -1235,19 +1415,19 @@ func (re *builtinRegexpReplaceFuncSig) evalString(row chunk.Row) (string, bool,
 		}
 
 		if re.isBinaryCollation() {
-			return re.getReplacedBinStr(reg, bexpr, trimmedBexpr, repl, pos, occurrence)
+			return re.getReplacedBinStr(reg, bexpr, trimmedBexpr, re.instructions, pos, occurrence)
 		}
-		return re.getReplacedStr(reg, expr, trimmedExpr, repl, trimmedLen, occurrence)
+		return re.getReplacedStr(reg, expr, trimmedExpr, re.instructions, trimmedLen+1, occurrence)
```

Comment:
Why change `trimmedLen` to `trimmedLen+1`, is this mean there is a bug in master branch?

Reply:
> Why change `trimmedLen` to `trimmedLen+1`, is this mean there is a bug in master branch?

No bug in master branch. In new implement, we need to pass position to the function, e.g. trimmed length is 3, and the start position is 4, so we need to change trimmedLen to trimmedLen+1. Maybe some parameter names is misleading, I will change it.